src/plugins/footnotes/index.ts

```diff
@@ -34,6 +34,7 @@ import { findTableInlineFootnoteIds } from './footnotes-utils'
 interface PluginState {
   nodes: [InlineFootnoteNode, number][]
   labels: Map<string, string>
+  tablesFootnoteLabels: Map<string, Map<string, number>>
```

Comment:
Given that you decided to use this same plugin for managing table footnotes - which is good- I would recommend using `nodes` &  `labels` to cover both instead of initiating this `tablesFootnoteLabels`. So what I'm suggesting is to simplify/refine what you are doing here by just pushing these table inline-notes and their labels when you build the state. And have a separate counter for the table footnotes. 

something close to this:
```
export const buildPluginState = (doc: ManuscriptNode): PluginState => {
  const nodes: [InlineFootnoteNode, number][] = []

  let index = 0
  let tableIndex = 0
  let previousTable: ContentNodeWithPos | undefined
  const labels = new Map<string, string>()

  doc.descendants((node, pos) => {
    if (isInlineFootnoteNode(node)) {
      nodes.push([node, pos])
      const $pos = doc.resolve(pos)
      const currentTable = findParentNodeClosestToPos(
        $pos,
        (node) => node.type === schema.nodes.table_element
      )
      if (currentTable) {
        if (previousTable && previousTable !== currentTable) {
          tableIndex = 0
        }
        node.attrs.rids.forEach((rid) => {
          labels.set(rid, String(++tableIndex))
        })
        previousTable = currentTable
      } else {
        node.attrs.rids.forEach((rid) => {
          labels.set(rid, String(++index))
        })
      }
    }
  })

  return { nodes, labels }
}
```
This way you will not need to do anything further after that.

Reply:
that will complicate things for us when we need to use that state, will need to rebuild it again! as with that you are mixing table footnote with the section footnote

Reply:
What's the problem with mixing them? Why do you need to rebuild it again? You'll just use the footnote ID to retrieve its label from the array, and that's it.

Reply:
I tried it locally we are going to end up with double labels for the table footnote, and fixing that will complicate the plugin more

Reply:
moved out of plugin